### PR TITLE
[FIX] l10n_sa_edi: setting tax scheme id to always be VAT

### PR DESCRIPTION
**Pull request by khfa-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by nim-odoo** [comment]
@robodoo priority

File: addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py [hunk 1]
```diff
@@ -185,9 +185,13 @@ def _get_partner_party_tax_scheme_vals_list(self, partner, role):
...
         """
         if role != 'customer' or partner.country_id.code == 'SA':
-            return super()._get_partner_party_tax_scheme_vals_list(partner, role)
+            vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
+            for vals in vals_list:
+                vals['tax_scheme_vals'] = {'id': 'VAT'}
+            return vals_list
         return []
 
+
```
Comment (Levizar) [inline addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py:194]:
```suggestion
```

**Comment by jco-odoo** [comment]
Ok, just awaiting update of commit msg

**Comment by nim-odoo** [comment]
@robodoo fw=skipci

**Review by Levizar — COMMENTED** [review]
To me, it is ok as it just put back the "VAT" as it was before done from ubl_2.0.
Just remove the extra line and I r+ it.

To me, it is very weird that in middle east format, they would put "NOT_EU" as they don't care.

CC @clbr-odoo @smetl

**Comment by nim-odoo** [comment]
@khfa-odoo Please fix the commit message name, thanks.

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Review by clbr-odoo — COMMENTED** [review]
@Levizar It actually make sense. We were hesitant with the previous fix for zatca because zatca is the only UBL format that does not seem to respect its rules.

UBL implements an EU standard, which has a rule that says that if the tax scheme is "VAT" the CompanyID must start with 2 letters, the country code. It's a rule that is followed by UBL, therefore should've been followed by all formats based on UBL, regardless of your country, but Zatca seems to be the exception :roll_eyes: 
We set "NOT_EU_VAT", but actually anything but "VAT" validates the rule.

Thank you for the fix.

**Pull request closed without merging**